zipline/data/minute_bars.py

```diff
@@ -1057,9 +1057,13 @@ def _open_minute_file(self, field, sid):
         try:
             carray = self._carrays[field][sid]
         except KeyError:
-            carray = self._carrays[field][sid] = \
-                bcolz.carray(rootdir=self._get_carray_path(sid, field),
-                             mode='r')
+            try:
+                carray = self._carrays[field][sid] = bcolz.carray(
+                    rootdir=self._get_carray_path(sid, field),
+                    mode='r',
+                )
+            except IOError:
+                raise NoDataOnDate('No minute data for sid {}.'.format(sid))
```

Comment:
Though technically there is no data for that date, this error really occurs when there is no data at all for the sid.
Should we have a different exception indicating that there is no data for any date, instead of `NoDataOnDate` which is just for the date requested?